pkg/reconciler/subscription/subscription.go

```diff
@@ -273,36 +273,40 @@ func (r *Reconciler) reconcile(ctx context.Context, subscription *v1alpha1.Subsc
 			zap.Error(err),
 			zap.Any("channel", subscription.Spec.Channel))
 		r.Recorder.Eventf(subscription, corev1.EventTypeWarning, channelReferenceFailed, "Failed to get Spec.Channel as Channelable duck type. %s", err)
-		subscription.Status.MarkChannelNotReady(channelReferenceFailed, "Failed to get Spec.Channel as Channelable duck type. %s", err)
+		subscription.Status.MarkChannelUnknown(channelReferenceFailed, "Failed to get Spec.Channel as Channelable duck type. %s", err)
 		return err
 	}
-	if err := r.subMarkedReadyByChannel(subscription, channel); err != nil {
+	ss, err := r.getSubStatusByChannel(subscription, channel)
+	if err != nil {
 		logging.FromContext(ctx).Warn("Subscription not marked by Channel as Ready.", zap.Error(err))
 		r.Recorder.Eventf(subscription, corev1.EventTypeWarning, subscriptionNotMarkedReadyByChannel, err.Error())
-		subscription.Status.MarkChannelNotReady(subscriptionNotMarkedReadyByChannel, "Subscription not marked by Channel as Ready: %s", err)
+		subscription.Status.MarkChannelFailed(subscriptionNotMarkedReadyByChannel, "Subscription not marked by Channel as Ready: %s", err)
```

Comment:
This error message is inaccurate. `subMarkedReadyByChannel` used to return nil only if the subscription was both present in the subscribers list and marked ready. Now the renamed `getSubStatusByChannel` returns nil if the subscription is present, regardless of readiness. Instead of `Subscription not marked by Channel as Ready`, this should be something more generic like `Failed to get subscription status`. 

This also seems like it should cause the channel state to be Unknown (like the above failures to get the channel) rather than Failed, but I haven't thought about it carefully.